Support tool lints with the `#[expect]` attribute (RFC 2383)

xFrednet opened the pull request:
This PR fixes the ICE https://github.com/rust-lang/rust/issues/94953 by making the assert for converted expectation IDs conditional.

Additionally, it moves the lint expectation check into a separate query to support rustdoc and other tools. On the way, I've also added some tests to ensure that the attribute works for Clippy and rustdoc lints.

The number of changes comes from the long test file. This may look like a monster PR, this may smell like a monster PR and this may be a monster PR, but it's a harmless monster. :sauropod: 

---

Closes: https://github.com/rust-lang/rust/issues/94953

cc: https://github.com/rust-lang/rust/issues/85549

r? @wesleywiser 

cc: @rust-lang/rustdoc

rust-highfive commented:
Some changes occurred in src/tools/clippy.

cc @rust-lang/clippy

flip1995 commented on compiler/rustc_middle/src/query/mod.rs:
```diff
@@ -147,6 +147,25 @@ rustc_queries! {
+    /// Tools can additionally pass in a tool filter. That will restrict the
+    /// expectations to only trigger for lints starting with the listed tool
+    /// name. This is useful for cases were not all linting code from rustc
+    /// was called. With the default `none` all registered lints will also
```
```suggestion
    /// was called. With the default `None` all registered lints will also
```

flip1995 reviewed (approved):
This LGTM. Though I'm not familiar with the query system and if just adding a new key for `Option<Symbol>` is ok, so I leave that to Wesley to decide.

xFrednet commented:
Hey @wesleywiser, could you take a look at this PR? :upside_down_face:

wesleywiser commented on src/test/ui/lint/rfc-2383-lint-reason/expect_tool_lint_rfc_2383.rs:
```diff
@@ -0,0 +1,154 @@
+// check-pass
+#![feature(lint_reasons)]
+
+//! This file tests the expect attribute implementation for tool lints. The same
+//! file will be used to test clippy and rustdoc.
```
It might be good to mention that this file is copied in three different places and we should try to keep them in sync with each other (and mention this in the other files as well).

xFrednet replied:
Good point, I thought that it was part of the comment, but it was missing now that you've mentioned it. :upside_down_face:

wesleywiser reviewed (approved):
Thanks for the ping @xFrednet! This looks really good and I'm glad to see we were able to resolve the ICE while keeping that check in place. I left one small suggestion with the tests added and then we can merge.

xFrednet commented:
I've rebased on master as well. I've only changed the mentioned comment and updated the line numbers in the `.stderr` files. Everything else stayed the same :upside_down_face:

wesleywiser commented:
Awesome! 

@bors r+

bors commented:
:pushpin: Commit c26742d979c359448e48ebf227017c7f5f1d783d has been approved by `wesleywiser`

<!-- @bors r=wesleywiser c26742d979c359448e48ebf227017c7f5f1d783d -->
<!-- homu: {"type":"Approved","sha":"c26742d979c359448e48ebf227017c7f5f1d783d","approver":"wesleywiser"} -->

bors commented:
:hourglass: Testing commit c26742d979c359448e48ebf227017c7f5f1d783d with merge 3076beb64cc568bead8924a7d63edf516ce1a662...
<!-- homu: {"type":"BuildStarted","head_sha":"c26742d979c359448e48ebf227017c7f5f1d783d","merge_sha":"3076beb64cc568bead8924a7d63edf516ce1a662"} -->

GuillaumeGomez commented:
Failed in https://github.com/rust-lang/rust/pull/96813:

```
 error: unresolved link to `unstable_expect_diagnostics`
   --> compiler/rustc_errors/src/lib.rs:429:15
    |
429 |     /// The [`unstable_expect_diagnostics`] should be empty when this struct is
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `unstable_expect_diagnostics` in scope
    |
    = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

error: unresolved link to `unstable_expect_diagnostics`
   --> compiler/rustc_errors/src/lib.rs:432:20
    |
432 |     /// check if [`unstable_expect_diagnostics`] is empty, if the expectation ids
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `unstable_expect_diagnostics` in scope
    |
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

error: could not document `rustc_errors`
```

@bors r-

xFrednet commented:
Thank you for the info. Weird that the PR CI didn't catch this. Are doc tests only run as part of bors? :thinking:

GuillaumeGomez commented:
Some checks are only run in bors indeed.

xFrednet commented:
I couldn't figure out how to link to another field in the same struct. I've now removed the links this is still sufficient, as the mentioned field is right below :upside_down_face:

GuillaumeGomez commented:
`Self::unstable_expect_diagnostics` should work.

xFrednet commented:
> Some checks are only run in bors indeed.

I think that this is the second time that this happens, the last time this was caught by a try run. Oh well :upside_down_face:

> `Self::unstable_expect_diagnostics` should work.

Rust analyzer doesn't highlight it :thinking:  I'll try to build the documentation locally and see (After I find the command for it xD)

GuillaumeGomez commented:
The command is (iirc): `x.py doc compiler --stage 1`.

GuillaumeGomez commented:
Also, try run does run more checks, however there wasn't one here, so I guess that's why you didn't see the bug. ;)

xFrednet commented:
> Also, try run does run more checks, however there wasn't one here, so I guess that's why you didn't see the bug. ;)

Jup, I've just added doc building to the CI of another project to catch these errors early. I'll be more careful about it from now on :upside_down_face: Thank you for the swift responses!

`Self::unstable_expect_diagnostics` does seam to work locally now, just the highlighting is a bit broken :)

![image](https://user-images.githubusercontent.com/17087237/167257090-a71e7326-7b21-4de1-8846-462ace8e4c40.png)

GuillaumeGomez commented:
Yeay!

@bors r=wesleywiser

bors commented:
:pushpin: Commit e23227c25073f7d2baa623d9db60e1f3f59c5031 has been approved by `wesleywiser`

<!-- @bors r=wesleywiser e23227c25073f7d2baa623d9db60e1f3f59c5031 -->
<!-- homu: {"type":"Approved","sha":"e23227c25073f7d2baa623d9db60e1f3f59c5031","approver":"wesleywiser"} -->

rust-log-analyzer commented:
A job failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/6334724038?check_suite_focus=true) [(plain)](https://github.com/rust-lang-ci/rust/commit/3076beb64cc568bead8924a7d63edf516ce1a662/checks/6334724038/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

```

</details>

bors commented:
:hourglass: Testing commit e23227c25073f7d2baa623d9db60e1f3f59c5031 with merge 31016be7a150fd3aeb8b49b55141ff5bb506a793...
<!-- homu: {"type":"BuildStarted","head_sha":"e23227c25073f7d2baa623d9db60e1f3f59c5031","merge_sha":"31016be7a150fd3aeb8b49b55141ff5bb506a793"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/6335548059?check_suite_focus=true)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/6335548059?check_suite_focus=true","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-aux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/6335404971?check_suite_focus=true) [(plain)](https://github.com/rust-lang-ci/rust/commit/31016be7a150fd3aeb8b49b55141ff5bb506a793/checks/6335404971/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

failures:

---- test_reverse::prop_reverse_headers stdout ----
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-122]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-138]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-151]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-167]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-174]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
error: test failed, to rerun pass '--test tests'
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-190]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-203]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-218]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-233]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-247]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-259]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-278]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-291]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-305]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-318]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-337]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-354]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-369]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-379]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-394]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-407]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-423]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-442]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-456]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-470]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-481]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-496]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-514]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-530]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-547]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-556]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-571]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-584]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-600]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-613]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-634]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-652]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-670]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-688]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-703]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-717]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'main' panicked at 'tests failed for https://github.com/BurntSushi/xsv', src/tools/cargotest/main.rs:101:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-729]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-750]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-764]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-778]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-797]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-810]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-824]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-837]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-850]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-864]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-879]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-891]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-907]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-926]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-940]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-952]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-966]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-991]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1003]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1018]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1031]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1047]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1066]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1084]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1093]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1108]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1124]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1141]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1155]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1169]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1184]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1194]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1207]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1217]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1230]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1240]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1250]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1264]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1278]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1289]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1300]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1312]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1327]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1337]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1349]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at 'assertion failed: `(left == right)`
  left: `[["]"], [""], ["獙9~"], ["f;텒"], [".\u{fff2}"], ["<\u{10ffff}"], ["\0\""], ["%"], ["9"], ["a\u{bb963}"], ["\u{14}"], ["Q+⁝\u{9f111}"]]`,
 right: `[["\u{feff}]"], [""], ["獙9~"], ["f;텒"], [".\u{fff2}"], ["<\u{10ffff}"], ["\0\""], ["%"], ["9"], ["a\u{bb963}"], ["\u{14}"], ["Q+⁝\u{9f111}"]]`', tests/test_reverse.rs:22:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1367]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1376]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1388]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at 'assertion failed: `(left == right)`
  left: `[["]"], ["%"], ["9"], ["a\u{bb963}"], ["\u{14}"], ["Q+⁝\u{9f111}"]]`,
 right: `[["\u{feff}]"], ["%"], ["9"], ["a\u{bb963}"], ["\u{14}"], ["Q+⁝\u{9f111}"]]`', tests/test_reverse.rs:22:5
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1397]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1412]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1423]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at 'assertion failed: `(left == right)`
  left: `[["]"], ["\u{14}"], ["Q+⁝\u{9f111}"]]`,
 right: `[["\u{feff}]"], ["\u{14}"], ["Q+⁝\u{9f111}"]]`', tests/test_reverse.rs:22:5
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1433]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1442]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1453]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at 'assertion failed: `(left == right)`
  left: `[["]"], ["\u{14}"]]`,
 right: `[["\u{feff}]"], ["\u{14}"]]`', tests/test_reverse.rs:22:5
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1467]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1481]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1490]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at 'assertion failed: `(left == right)`
  left: `[["]"]]`,
 right: `[["\u{feff}]"]]`', tests/test_reverse.rs:22:5
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1504]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1513]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1527]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1536]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at 'assertion failed: `(left == right)`
  left: `[]`,
 right: `[["\u{feff}"]]`', tests/test_reverse.rs:22:5
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1544]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1555]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1564]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1573]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1584]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1594]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1603]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1614]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1622]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1629]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1640]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1654]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1657]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1667]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1675]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1682]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
[/checkout/obj/build/ct/xsv/target/debug/xit/prop_reverse_headers/test-1689]: "/checkout/obj/build/ct/xsv/target/debug/xsv" "reverse" "in.csv"
thread 'test_reverse::prop_reverse_headers' panicked at '[quickcheck] TEST FAILED (runtime error). Arguments: (CsvData { data: [[[239, 187, 191]]] })
Error: "assertion failed: `(left == right)`\n  left: `[]`,\n right: `[[\"\\u{feff}\"]]`"', /cargo/registry/src/github.com-1ecc6299db9ec823/quickcheck-0.7.1/src/tester.rs:176:28

failures:
    test_reverse::prop_reverse_headers


test result: FAILED. 425 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.94s

Build completed unsuccessfully in 0:25:13
make: *** [check-aux] Error 1
Makefile:44: recipe for target 'check-aux' failed
```

</details>

compiler-errors commented:
Bors sync made this re-approved, sorry for the noise

@bors r-

xFrednet commented:
The error message from @rust-log-analyzer seems weird to me. IDK what the error is and where it could come from, my guess/hope is that this was just some CI weirdness. I've rebased on master, that should (hopefully) resolve everything. The commits remain unchanged :upside_down_face:

GuillaumeGomez commented:
Let's hope so. :wink:  I'll let you handle the r+ so you don't need to ping me or @wesleywiser again when CI pass. Please use `@bors r=wesleywiser` once it's green.

@bors delegate=xFrednet

bors commented:
:pushpin: Commit 332a9fa200f6e36fb8d5fbaf8e9fe0433562beea has been approved by `wesleywiser``

<!-- @bors r=wesleywiser` 332a9fa200f6e36fb8d5fbaf8e9fe0433562beea -->
<!-- homu: {"type":"Approved","sha":"332a9fa200f6e36fb8d5fbaf8e9fe0433562beea","approver":"wesleywiser`"} -->

GuillaumeGomez commented:
*sigh* I didn't expect that one...

@bors r-
@bors delegate=xFrednet

bors commented:
:v: @xFrednet can now approve this pull request
<!-- homu: {"type":"Delegated","delegator":"GuillaumeGomez","delegate":"xFrednet"} -->

xFrednet commented:
Thanks, I thought about using r=wesleywiser before, but was unsure if that would be okay :sweat_smile:

@bors r=wesleywiser

bors commented:
:pushpin: Commit 332a9fa200f6e36fb8d5fbaf8e9fe0433562beea has been approved by `wesleywiser`

<!-- @bors r=wesleywiser 332a9fa200f6e36fb8d5fbaf8e9fe0433562beea -->
<!-- homu: {"type":"Approved","sha":"332a9fa200f6e36fb8d5fbaf8e9fe0433562beea","approver":"wesleywiser"} -->

bors commented:
:hourglass: Testing commit 332a9fa200f6e36fb8d5fbaf8e9fe0433562beea with merge 7e33efee9b7aba31953af392753e6f8bb682bd82...
<!-- homu: {"type":"BuildStarted","head_sha":"332a9fa200f6e36fb8d5fbaf8e9fe0433562beea","merge_sha":"7e33efee9b7aba31953af392753e6f8bb682bd82"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/6341440554?check_suite_focus=true)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/6341440554?check_suite_focus=true","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-stable`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/6341258917?check_suite_focus=true) [(plain)](https://github.com/rust-lang-ci/rust/commit/7e33efee9b7aba31953af392753e6f8bb682bd82/checks/6341258917/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] rustc_builtin_macros test:false 1.657
error: unresolved link to `analysis`
   --> compiler/rustc_middle/src/ty/query.rs:213:17
    |
213 |             $($(#[$attr])* pub $name: query_storage::$name<$tcx>,)*
...
335 | rustc_query_append! { [define_callbacks!][<'tcx>] }
    | --------------------------------------------------- in this macro invocation
    |
    |
    = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
    = note: the link appears in this line:
            
            only execute a small subset of the [`analysis`] query, while allowing
    = note: no item named `analysis` in scope
    = note: no item named `analysis` in scope
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
    = note: this error originates in the macro `define_callbacks` (in Nightly builds, run with -Z macro-backtrace for more info)
error: unresolved link to `analysis`
   --> compiler/rustc_middle/src/ty/query.rs:217:17
    |
    |
217 |             $($(#[$attr])*
...
335 | rustc_query_append! { [define_callbacks!][<'tcx>] }
    | --------------------------------------------------- in this macro invocation
    |
    |
    = note: the link appears in this line:
            
            only execute a small subset of the [`analysis`] query, while allowing
    = note: no item named `analysis` in scope
    = note: no item named `analysis` in scope
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
    = note: this error originates in the macro `define_callbacks` (in Nightly builds, run with -Z macro-backtrace for more info)
error: unresolved link to `analysis`
   --> compiler/rustc_middle/src/ty/query.rs:235:17
    |
    |
235 |             $($(#[$attr])*
...
335 | rustc_query_append! { [define_callbacks!][<'tcx>] }
    | --------------------------------------------------- in this macro invocation
    |
    |
    = note: the link appears in this line:
            
            only execute a small subset of the [`analysis`] query, while allowing
    = note: no item named `analysis` in scope
    = note: no item named `analysis` in scope
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
    = note: this error originates in the macro `define_callbacks` (in Nightly builds, run with -Z macro-backtrace for more info)
error: unresolved link to `analysis`
   --> compiler/rustc_middle/src/ty/query.rs:245:17
    |
    |
245 |             $($(#[$attr])*
...
335 | rustc_query_append! { [define_callbacks!][<'tcx>] }
    | --------------------------------------------------- in this macro invocation
    |
    |
    = note: the link appears in this line:
            
            only execute a small subset of the [`analysis`] query, while allowing
    = note: no item named `analysis` in scope
    = note: no item named `analysis` in scope
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
    = note: this error originates in the macro `define_callbacks` (in Nightly builds, run with -Z macro-backtrace for more info)
error: unresolved link to `analysis`
   --> compiler/rustc_middle/src/ty/query.rs:311:17
    |
    |
311 |             $($(#[$attr])*
...
335 | rustc_query_append! { [define_callbacks!][<'tcx>] }
    | --------------------------------------------------- in this macro invocation
    |
    |
    = note: the link appears in this line:
            
            only execute a small subset of the [`analysis`] query, while allowing
    = note: no item named `analysis` in scope
    = note: no item named `analysis` in scope
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
    = note: this error originates in the macro `define_callbacks` (in Nightly builds, run with -Z macro-backtrace for more info)
error: could not document `rustc_middle`

Caused by:
Caused by:
  process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustdoc --edition=2021 --crate-type lib --crate-name rustc_middle compiler/rustc_middle/src/lib.rs --target x86_64-unknown-linux-gnu -o /checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/doc -Zunstable-options --check-cfg 'values(feature, "rustc-rayon", "rustc-rayon-core", "rustc_use_parallel_compiler")' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -C metadata=ef905fb68a932e45 -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/release/deps --extern bitflags=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libbitflags-ca31efbac0a66613.rmeta --extern chalk_ir=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libchalk_ir-792fdec427bd0208.rmeta --extern either=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libeither-91b390aa12a1759b.rmeta --extern gsgdt=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libgsgdt-4a86e56ba107250d.rmeta --extern polonius_engine=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libpolonius_engine-756901c8eb76ec02.rmeta --extern rand=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librand-552e6be05858db2b.rmeta --extern rand_xoshiro=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librand_xoshiro-3001b9d83692bf5a.rmeta --extern rustc_apfloat=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_apfloat-ca8191ee5bf99b0c.rmeta --extern rustc_arena=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_arena-7ba6b4dbcf7ca5ae.rmeta --extern rustc_ast=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_ast-5ad3a355308983d0.rmeta --extern rustc_attr=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_attr-fdb613daff2ce5ab.rmeta --extern rustc_data_structures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_data_structures-a7f711b2b3b6a088.rmeta --extern rustc_errors=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_errors-8a848e6444b14b7f.rmeta --extern rustc_feature=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_feature-702d6c98e24c6085.rmeta --extern rustc_graphviz=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_graphviz-eacc3a5f9f5bcdda.rmeta --extern rustc_hir=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_hir-d1897ef46668ab67.rmeta --extern rustc_index=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_index-b4c81c5228e1a608.rmeta --extern rustc_macros=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/release/deps/librustc_macros-bd658f125c9c41eb.so --extern rustc_query_system=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_query_system-7873a0697a8100e9.rmeta --extern rustc_serialize=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_serialize-15acdc7410a458fc.rmeta --extern rustc_session=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_session-d6c2a336abbc7f92.rmeta --extern rustc_span=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_span-5aea05b1b10fb07b.rmeta --extern rustc_target=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_target-69c636f45962117f.rmeta --extern rustc_type_ir=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_type_ir-462114c2a778f1ab.rmeta --extern smallvec=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libsmallvec-00b7d367e0374797.rmeta --extern tracing=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-rustc/x86_64-unknown-linux-gnu/release/deps/libtracing-b4b3f0fc8c99da0f.rmeta --extern-html-root-url 'bitflags=https://docs.rs/bitflags/1.2.1/' --extern-html-root-url 'chalk_ir=https://docs.rs/chalk-ir/0.80.0/' --extern-html-root-url 'either=https://docs.rs/either/1.6.0/' --extern-html-root-url 'gsgdt=https://docs.rs/gsgdt/0.1.2/' --extern-html-root-url 'polonius_engine=https://docs.rs/polonius-engine/0.13.0/' --extern-html-root-url 'rand=https://docs.rs/rand/0.8.5/' --extern-html-root-url 'rand_xoshiro=https://docs.rs/rand_xoshiro/0.6.0/' --extern-html-root-url 'smallvec=https://docs.rs/smallvec/1.7.0/' --extern-html-root-url 'tracing=https://docs.rs/tracing/0.1.29/' -Zunstable-options -Csymbol-mangling-version=v0 -Zunstable-options '--check-cfg=names()' '--check-cfg=values()' '--check-cfg=values(bootstrap)' '--check-cfg=values(parallel_compiler)' '--check-cfg=values(no_btreemap_remove_entry)' '--check-cfg=values(crossbeam_loom)' '--check-cfg=values(span_locations)' -Dwarnings '-Wrustdoc::invalid_codeblock_attributes' --crate-version '1.62.0
  (7e33efee9
  2022-05-08)' --document-private-items '-Arustdoc::private-intra-doc-links' --enable-index-page -Zunstable-options -Znormalize-docs --show-type-layout --generate-link-to-definition --extern-html-root-url 'ena=https://docs.rs/ena/latest/'` (exit status: 1)
[RUSTC-TIMING] rustc_middle test:false 23.595
Build completed unsuccessfully in 0:33:31
```

</details>

xFrednet commented:
@bors r-

xFrednet commented:
Let's hope that third times is the charm.

@bors r=wesleywiser rollup=iffy

bors commented:
:pushpin: Commit 9516a40f1e767d1181344f9a3df8f740dff673c5 has been approved by `wesleywiser`

<!-- @bors r=wesleywiser 9516a40f1e767d1181344f9a3df8f740dff673c5 -->
<!-- homu: {"type":"Approved","sha":"9516a40f1e767d1181344f9a3df8f740dff673c5","approver":"wesleywiser"} -->

bors commented:
:hourglass: Testing commit 9516a40f1e767d1181344f9a3df8f740dff673c5 with merge a57117982aa99f410f6efcba511998c1161adfb8...
<!-- homu: {"type":"BuildStarted","head_sha":"9516a40f1e767d1181344f9a3df8f740dff673c5","merge_sha":"a57117982aa99f410f6efcba511998c1161adfb8"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/6345042501?check_suite_focus=true)
Approved by: wesleywiser
Pushing a57117982aa99f410f6efcba511998c1161adfb8 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"wesleywiser","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/6345042501?check_suite_focus=true"},"merge_sha":"a57117982aa99f410f6efcba511998c1161adfb8"} -->

rust-timer commented:
Finished benchmarking commit (a57117982aa99f410f6efcba511998c1161adfb8): [comparison url](https://perf.rust-lang.org/compare.html?start=cb121987158d69bb894ba1bcc21dc45d1e0a488f&end=a57117982aa99f410f6efcba511998c1161adfb8).

**Summary**: This benchmark run did not return any relevant results.

If you disagree with this performance assessment, please file an issue in [rust-lang/rustc-perf](https://github.com/rust-lang/rustc-perf/issues/new).



@rustbot label: -perf-regression
